cores/beken-72xx/arduino/libraries/WiFi/WiFiSTA.cpp

```diff
@@ -246,3 +246,39 @@ WiFiAuthMode WiFiClass::getEncryption() {
 	STA_GET_LINK_STATUS_RETURN(WIFI_AUTH_INVALID);
 	return securityTypeToAuthMode(LINK_STATUS.security);
 }
+#ifdef CONFIG_IPV6
+bool WiFiClass::enableIpV6() {
+	return true;
+}
+
+IPv6Address WiFiClass::localIPv6() {
+	struct netif *ifs = (struct netif *)net_get_sta_handle();
+	std::vector<IPv6Address> result;
+	struct wlan_ip_config addr;
+	int nr_addresses = 0;
+
+	if (sta_ip_is_start())
+		nr_addresses = net_get_if_ipv6_pref_addr(&addr, ifs);
+
+	for (int i = 0; i < nr_addresses; i++)
+		if (ip6_addr_islinklocal(&addr.ipv6[i]))
+			return IPv6Address(addr.ipv6[i].addr);
```

Comment:
I'd prefer having braces around loops:

```suggestion
	for (int i = 0; i < nr_addresses; i++) {
		if (ip6_addr_islinklocal(&addr.ipv6[i]))
			return IPv6Address(addr.ipv6[i].addr);
	}
```